src/core/column/reduce_unary.h

```diff
@@ -26,19 +26,20 @@
 namespace dt {
 
 
-template <typename T, bool IS_GROUPED>
+template <typename T_IN, typename T_OUT, bool IS_GROUPED>
 class ReduceUnary_ColumnImpl : public Virtual_ColumnImpl {
   protected:
     Column col_;
     Groupby gby_;
+    
 
   public:
     ReduceUnary_ColumnImpl(Column &&col, const Groupby& gby)
-      : Virtual_ColumnImpl(gby.size(), col.stype()),
+      : Virtual_ColumnImpl(gby.size(), stype_from<T_OUT>),
```

Comment:
You need to pass `stype` here just like it was before, otherwise it is not possible to properly get it from `T_OUT`. I'm not sure what was the reason to change it?

Reply:
the change was made here to allow passing int64 for count, instead of casting say a float to int64, as we did for the other reducers so far(sum/prod/mean). For min/max, we did not need to cast since it is within the same data type. Another option would be to pass an explicit stype here, instead of deriving the stype from T_OUT

Reply:
Well, but the resulting column for `count()` has `int64` stype, so `col.stype()` will work in this case too. Anyways, `stype_from<T_OUT>` can only give you a guess.

Reply:
why does `stype_from<T_OUT>` not work properly? maybe something I am missing from my interpretation of the function? or is `Type::from` d right way to go?

Reply:
The way to go is to use `col.stype()` as it was before..

It will not work because there is no strict way to get stype from the underlying type. We may have different column to have the same C++ type: date32/int32, time64/int64, boo8/int8, etc.

Reply:
so it is ok to cast say a float64 to int64 for the count() function? no need to worry about precision?

Reply:
But why do we need to cast it for count?

Reply:
All reducers are separate classes, casting was needed for some of them outside of this PR. What I say is that you need to restore the original implementation for those reducers, otherwise it will not work with only passing the C++ type.

Note, tests are failing not for count…